src/microbe_stage/MembraneWaterRipple.cs

```
@@ -0,0 +1,572 @@
+        {
+            visibilityCheckTimer = 0.0f;
+            UpdateCameraCache();
+            isCurrentlyVisible = IsMembraneVisible();
+
+            // Toggle water plane visibility
+            if (waterPlane.Visible != (isCurrentlyVisible && isEffectEnabled))
+            {
+                waterPlane.Visible = isCurrentlyVisible && isEffectEnabled;
+            }
+        }
+
+        if (!isCurrentlyVisible || !isEffectEnabled)
+            return;
+
+        UpdatePosition();
+
+        // Update time for animation based on LOD level and movement
+        float timeScale = 1.0f;
+        if (currentSpeed > 0.1f)
+        {
+            // Scale time update by current LOD level
+            switch (currentLodLevel)
+            {
+                case LodLevel.VeryLow:
+                    timeScale = 1.0f + currentSpeed * 0.05f;
+                    break;
+
+                case LodLevel.Low:
+                    timeScale = 1.0f + currentSpeed * 0.08f;
+                    break;
+
+                case LodLevel.Medium:
+                    timeScale = 1.0f + currentSpeed * 0.1f;
+                    break;
+
+                case LodLevel.High:
+                    timeScale = 1.0f + currentSpeed * 0.12f;
+                    break;
+            }
+        }
+
+        timeAccumulator += (float)delta * timeScale;
+        waterMaterial?.SetShaderParameter(timeOffsetParam, timeAccumulator);
+
+        // Update movement parameters (with reduced frequency for distant objects)
+        if (currentLodLevel == LodLevel.VeryLow)
+        {
+            // For very distant membranes, update less frequently
+            if ((int)(timeAccumulator * 10) % 3 == 0)
+            {
+                UpdateMovementParameters((float)delta);
+            }
+        }
+        else
+        {
+            UpdateMovementParameters((float)delta);
+        }
+    }
+
+    private void InitializeShaderParameters()
+    {
+        if (waterMaterial == null)
```

Water material seems important enough to make it non-nullable and then throw big errors in _Ready if it is missing.

Suppressing errors without even a GD.PrintErr is almost the worst thing to do as someone will run into a bug like this at some point and then we'll be totally clueless as to what went wrong.